src/test/java/org/opensearch/knn/index/LuceneEngineIT.java

```diff
@@ -0,0 +1,270 @@
+
+        refreshAllIndices();
+        assertEquals(1, getDocCount(INDEX_NAME));
+    }
+
+    public void testDeleteDoc() throws Exception {
+        createKnnIndexMappingWithLuceneEngine(2, SpaceType.L2);
+        Float[] vector = { 6.0f, 6.0f };
+        addKnnDoc(INDEX_NAME, DOC_ID, FIELD_NAME, vector);
+
+        deleteKnnDoc(INDEX_NAME, DOC_ID);
+
+        refreshAllIndices();
+        assertEquals(0, getDocCount(INDEX_NAME));
+    }
+
+    private void createKnnIndexMappingWithLuceneEngine(int dimension, SpaceType spaceType) throws Exception {
+        XContentBuilder builder = XContentFactory.jsonBuilder()
+            .startObject()
+            .startObject("properties")
+            .startObject(FIELD_NAME)
+            .field("type", "knn_vector")
+            .field("dimension", dimension)
+            .startObject(KNNConstants.KNN_METHOD)
+            .field(KNNConstants.NAME, KNNEngine.LUCENE.getMethod(METHOD_HNSW).getMethodComponent().getName())
+            .field(KNNConstants.METHOD_PARAMETER_SPACE_TYPE, spaceType.getValue())
+            .field(KNNConstants.KNN_ENGINE, KNNEngine.LUCENE.getName())
+            .startObject(KNNConstants.PARAMETERS)
+            .field(KNNConstants.METHOD_PARAMETER_M, M)
+            .field(KNNConstants.METHOD_PARAMETER_EF_CONSTRUCTION, EF_CONSTRUCTION)
+            .endObject()
+            .endObject()
+            .endObject()
+            .endObject()
+            .endObject();
+
+        String mapping = Strings.toString(builder);
+        createKnnIndex(INDEX_NAME, mapping);
+    }
+
+    private void baseQueryTest(SpaceType spaceType) throws Exception {
+
+        createKnnIndexMappingWithLuceneEngine(DIMENSION, spaceType);
+        for (int j = 0; j < TEST_INDEX_VECTORS.length; j++) {
+            addKnnDoc(INDEX_NAME, Integer.toString(j + 1), FIELD_NAME, TEST_INDEX_VECTORS[j]);
+        }
+
+        validateQueries(spaceType, FIELD_NAME);
+    }
+
+    private void validateQueries(SpaceType spaceType, String fieldName) throws IOException {
+
+        int k = LuceneEngineIT.TEST_INDEX_VECTORS.length;
+        for (float[] queryVector : TEST_QUERY_VECTORS) {
+            Response response = searchKNNIndex(INDEX_NAME, new KNNQueryBuilder(fieldName, queryVector, k), k);
+            String responseBody = EntityUtils.toString(response.getEntity());
+            List<KNNResult> knnResults = parseSearchResponse(responseBody, fieldName);
+            assertEquals(k, knnResults.size());
+
+            List<Float> actualScores = parseSearchResponseScore(responseBody, fieldName);
+            for (int j = 0; j < k; j++) {
+                float[] primitiveArray = Floats.toArray(Arrays.stream(knnResults.get(j).getVector()).collect(Collectors.toList()));
+                float distance = TestUtils.computeDistFromSpaceType(spaceType, primitiveArray, queryVector);
+                float rawScore = VECTOR_SIMILARITY_TO_SCORE.get(spaceType.getVectorSimilarityFunction()).apply(distance);
```

Comment:
this change is required to address breaking change in Lucene 9.3. Method _convertToScore_
existis in [Lucene 9.2](
https://github.com/apache/lucene/blob/branch_9_2/lucene/core/src/java/org/apache/lucene/index/VectorSimilarityFunction.java#L108) but has been removed in [Lucene 9.3](https://github.com/apache/lucene/blob/branch_9_3/lucene/core/src/java/org/apache/lucene/index/VectorSimilarityFunction.java)